src/ppx/xr_component.cpp

```diff
@@ -0,0 +1,395 @@
+{
+    const char* graphicsAPIExtension = "";
+    // Extensions (Non-Optional)
+#if defined(PPX_D3D11)
+    graphicsAPIExtension = XR_KHR_D3D11_ENABLE_EXTENSION_NAME;
+#endif // defined(PPX_D3D11)
+
+#if defined(PPX_D3D12)
+    graphicsAPIExtension = XR_KHR_D3D12_ENABLE_EXTENSION_NAME;
+#endif // defined(PPX_D3D12)
+
+#if defined(PPX_VULKAN)
+    graphicsAPIExtension = XR_KHR_VULKAN_ENABLE2_EXTENSION_NAME;
+#endif // defined(PPX_VULKAN)
+
+    std::vector<const char*> xrInstanceExtensions;
+    xrInstanceExtensions.push_back(graphicsAPIExtension);
+    if (enableDebug) {
+        xrInstanceExtensions.push_back(XR_EXT_DEBUG_UTILS_EXTENSION_NAME);
+    }
+    // Verify extensions
+    uint32_t extCount = 0;
+    xrEnumerateInstanceExtensionProperties(nullptr, 0, &extCount, nullptr);
+    std::vector<XrExtensionProperties> xrExts(extCount, {XR_TYPE_EXTENSION_PROPERTIES});
+    xrEnumerateInstanceExtensionProperties(nullptr, extCount, &extCount, xrExts.data());
+    for (const auto& ext : xrInstanceExtensions) {
+        auto it = std::find_if(xrExts.begin(), xrExts.end(), [&](const XrExtensionProperties& e) { return strcmp(e.extensionName, ext) == 0; });
+        PPX_ASSERT_MSG((it != xrExts.end()), "The extension is not supported! Also make sure your OpenXR runtime is loaded properly!");
+    }
+
+    // Layers (Optional)
+    std::vector<const char*> xrRequestedInstanceLayers;
+    if (enableDebug) {
+        // This requires adding following lines in Environment in Debugging Settings
+        // XR_ENABLE_API_LAYERS=XR_APILAYER_LUNARG_core_validation
+        // XR_API_LAYER_PATH= *folder that contains XrApiLayer_core_validation.json*
+        xrRequestedInstanceLayers.push_back("XR_APILAYER_LUNARG_core_validation");
+    }
+    uint32_t layerCount;
+    xrEnumerateApiLayerProperties(0, &layerCount, nullptr);
+    std::vector<XrApiLayerProperties> xrSupportedLayers(layerCount, {XR_TYPE_API_LAYER_PROPERTIES});
+    xrEnumerateApiLayerProperties(layerCount, &layerCount, xrSupportedLayers.data());
+
+    std::vector<const char*> xrInstanceLayers;
+
+    for (const auto& layer : xrRequestedInstanceLayers) {
+        auto it = std::find_if(xrSupportedLayers.begin(), xrSupportedLayers.end(), [&](const XrApiLayerProperties& l) { return strcmp(l.layerName, layer) == 0; });
+        if (it != xrSupportedLayers.end()) {
+            xrInstanceLayers.push_back(layer);
+        }
+    }
+
+    // Create XrInstance
+    XrInstanceCreateInfo createInfo       = {XR_TYPE_INSTANCE_CREATE_INFO};
+    createInfo.enabledExtensionCount      = (uint32_t)xrInstanceExtensions.size();
+    createInfo.enabledExtensionNames      = xrInstanceExtensions.data();
+    createInfo.enabledApiLayerCount       = (uint32_t)xrInstanceLayers.size();
+    createInfo.enabledApiLayerNames       = xrInstanceLayers.data();
+    createInfo.applicationInfo.apiVersion = XR_CURRENT_API_VERSION;
+    strcpy_s(createInfo.applicationInfo.applicationName, "VR Test");
```

Comment:
The application name should come from the constructor

Reply:
fixed